python_agent/test_custom_env.py

```
@@ -0,0 +1,154 @@
+        # Ensure last_observation is set and has no bones
+        empty_obs_data = self._get_default_raw_obs()
+        self.mock_zmq_client.get_observation.return_value = empty_obs_data # for _get_obs calls
+        self.env.last_observation = self.env._get_obs() # Populate last_observation with no bones
+
+        self.mock_zmq_client.execute_action.reset_mock() # Reset call count
+        _, reward, _, _, _ = self.env.step(5) # Action 5 = pickup_bone
+        
+        self.mock_zmq_client.execute_action.assert_not_called()
+        self.assertEqual(reward, -1.0) # Penalty for trying to pick up non-visible bones
+
+    def test_step_pickup_bone_bones_visible(self):
```

Ruff (0.11.9)

Missing return type annotation for public function `test_step_pickup_bone_bones_visible`

Add return type annotation: `None`

(ANN201)